components/cosmetic_filters/renderer/cosmetic_filters_js_render_frame_observer.cc

```diff
@@ -0,0 +1,59 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/cosmetic_filters/renderer/cosmetic_filters_js_render_frame_observer.h"
+
+#include "base/bind.h"
+#include "content/public/renderer/render_frame.h"
+#include "third_party/blink/public/web/web_local_frame.h"
+
+namespace cosmetic_filters {
+
+CosmeticFiltersJsRenderFrameObserver::CosmeticFiltersJsRenderFrameObserver(
+    content::RenderFrame* render_frame,
+    const int32_t isolated_world_id)
+    : RenderFrameObserver(render_frame),
+      isolated_world_id_(isolated_world_id) {}
+
+CosmeticFiltersJsRenderFrameObserver::~CosmeticFiltersJsRenderFrameObserver() {}
+
+void CosmeticFiltersJsRenderFrameObserver::DidStartNavigation(
+    const GURL& url,
+    base::Optional<blink::WebNavigationType> navigation_type) {
+  url_ = url;
```

Comment:
not sure why do we need this method, can we just extract `render_frame()->GetWebFrame()->GetDocument().Url()` or something in `DidCreateNewDocument()` ?

Basically wondering whether `DidStartNavigation()` is called for redirects, otherwise we can cache a wrong url

Reply:
it does called for redirects. This is basically done as DOM distiller does at the RenderFrameObserver, they get URL from this method as well
`void DistillerJsRenderFrameObserver::DidStartNavigation(`
`    const GURL& url,`
`    base::Optional<blink::WebNavigationType> navigation_type) {`
`  is_distiller_page_ = url_utils::IsDistilledPage(url);`
`}`